app/models/hotel.rb

```diff
@@ -1,3 +1,7 @@
 class Hotel < ApplicationRecord
    has_many :rooms
+
+   def guests
+      Guest.joins(:guest_rooms).joins(:rooms).where("rooms.hotel_id = #{self.id}").distinct
+   end
 end
```

Comment:
I really the `guests` method here. What if I told you there was an easier way to do the exact same logic via an association?

If we take a look at the `Hotel` model, we'll see that we have one association between a `Hotel` and a `Room`. We also know that if we take a look at the `Room` model, a `Room` has an association with `Guest`s through `GuestRoom`.

There's a neat trick in Rails that you can use those predefined associations to simplify our joins.

```ruby
class Hotel < ApplicationRecord
  has_many :rooms
  has_many :guests, through: :rooms # We are gaining access to guests of this hotel through the rooms of this hotel

  def all_guests
    guests.distinct
  end
end
```

Rails magic FTW. Play around with this and see what SQL gets printed in the terminal. A multi-join just like you defined by hand in your own logic!